src/App.js

```diff
@@ -0,0 +1,33 @@
+import React, { Component } from 'react';
+import './App.css';
+
+class App extends Component {
+  state = {
+    tasks: ['something'],
+    value: '',
+  }
+
+  addTask = () => this.setState(() => !!this.state.value.trimRight() && !this.state.tasks.includes(this.state.value) ? this.state.tasks.push(this.state.value) : null)
+
+  removeTask = idToRemove => this.setState({tasks: this.state.tasks.filter((task, id) => id !== idToRemove )})
+
+  render() {
+    return (
+      <div className="app">
+        <div className="title">ToDo App!</div>
+        <div className="textInput-block">
+          <label>Input task:</label>
+          <input onChange={ key => this.setState({value: key.target.value})} value={this.state.value}></input>
```

Comment:
we should extract function `key => ...` to this class method for performance reason. Currently - this function will be created each times component renders. If you put it as a method, it will be created only once (during mounting).

Reply:
argument which is passed to this function is an event object actually.